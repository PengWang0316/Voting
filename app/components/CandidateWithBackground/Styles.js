import { StyleSheet } from 'react-native';

import Theme from '../../Theme';

export default StyleSheet.create({
  root: {
    width: '100%',
    height: '50%',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  democracyBg: {
    backgroundColor: '#1a65dd',
  },
  republicBg: {
    backgroundColor: '#dd201a',
  },
  avatar: {
    width: 200,
    height: 200,
    borderRadius: 100,
    marginBottom: 20,
  },
  voteView: {
    // flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  voteText: {
    fontWeight: 'bold',
    fontSize: 20,
    color: Theme.primary.contrastText,
    marginLeft: 10,
  },
});
