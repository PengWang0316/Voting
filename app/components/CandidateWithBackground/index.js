import CandidateWithBackground from './CandidateWithBackground';

export default CandidateWithBackground;
