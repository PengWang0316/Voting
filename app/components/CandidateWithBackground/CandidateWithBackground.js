import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import CacheableImage from 'react-native-cacheable-image';
import { Icon } from 'react-native-elements';

import Styles from './Styles';

type Props = {
  candidate: Object,
  onClick: Function,
};

export const CandidateWithBackground = ({ candidate, onClick }: Props) => (
  <View
    style={[Styles.root, candidate.isDemocracy ? Styles.democracyBg : Styles.republicBg]}
  >
    <TouchableOpacity onPress={() => onClick(candidate)} style={Styles.root}>
      <CacheableImage
        style={Styles.avatar}
        source={{ uri: candidate.photo }}
      />
      <View style={Styles.voteView}>
        <Icon
          name="tag-heart"
          type="material-community"
          color="white"
        />
        <Text style={Styles.voteText}>{candidate.votes}</Text>
      </View>
    </TouchableOpacity>
  </View>
);
export default CandidateWithBackground;
