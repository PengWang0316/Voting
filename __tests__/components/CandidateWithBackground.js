import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { CandidateWithBackground } from '../../app/components/CandidateWithBackground/CandidateWithBackground';

jest.mock('react-native-cacheable-image', () => 'CacheableImage');
jest.mock('react-native', () => ({
  View: 'View',
  TouchableOpacity: 'TouchableOpacity',
  StyleSheet: {
    create: jest.fn().mockReturnValue({
      root: {
        width: '100%',
        height: '50%',
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
      },
      democracyBg: {
        backgroundColor: '#1a65dd',
      },
      republicBg: {
        backgroundColor: '#dd201a',
      },
      avatar: {
        width: 200,
        height: 200,
        borderRadius: 100,
      },
    }),
  },
}));

describe('CandidateWithBackground', () => {
  const defaultProps = {
    candidate: {
      id: 1,
      isDemocracy: 1,
      photo: 'photoUri',
    },
    onClick: jest.fn(),
  };
  const getShallowComponent = (props = defaultProps) => shallow(<CandidateWithBackground {...props} />);

  test('Click TouchableOpacity', () => {
    const component = getShallowComponent();
    component.find('TouchableOpacity').simulate('press');
    expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
    expect(defaultProps.onClick).toHaveBeenLastCalledWith(defaultProps.candidate);
  });

  test('Snapshot', () => expect(renderer.create(<CandidateWithBackground {...defaultProps} />).toJSON()).toMatchSnapshot());
});
