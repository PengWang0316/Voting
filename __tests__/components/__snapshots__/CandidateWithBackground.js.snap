// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidateWithBackground Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "50%",
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "backgroundColor": "#1a65dd",
      },
    ]
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "50%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <CacheableImage
      source={
        Object {
          "uri": "photoUri",
        }
      }
      style={
        Object {
          "borderRadius": 100,
          "height": 200,
          "width": 200,
        }
      }
    />
    <View>
      <Icon
        color="white"
        name="tag-heart"
        type="material-community"
      />
      <Text>
        100
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
